sage.py

```diff
@@ -481,1274 +423,920 @@ def getepisodes(action_seq, mcat, plot, debug=False):
-            samerank += '"}'
-            lines.append((0,samerank))
-
+            # All obj variants have the same rank
+            lines.append('{ rank=same; "' + '" "'.join([_translate(obj) for obj in observed_obj]) + '"}')
 
+            node_signatures = dict()
             already_addressed = set()
-            for attackerID,attempts in team_level.items(): # for every attacker that obtains this objective
-                color = tcols[attackerID.split('-')[0]] # team color
-                ones = [''.join([action[0] for action in attempt]) for attempt in attempts]
-                unique = len(set(ones)) # count exactly unique attempts
-                #print(unique)
-                #print('team', attackerID, 'total paths', len(attempts), 'unique paths', unique, 'longest path:', max([len(x) for x in attempts]), \
-                #     'shortest path:', min([len(x) for x in attempts]))
-                
-                #path_info[attack][attackerID].append((len(attempts), unique, max([len(x) for x in attempts]), min([len(x) for x in attempts])))
-                
-                for attempt in attempts: # iterate over each attempt
-                    # record all nodes
-                    for action in attempt:
-                        if action[0] not in nodes.keys():
-                            nodes[action[0]] = set()
-                        nodes[action[0]].update(action[3])
-                    # nodes
-                    for vid,(vname,start_time,end_time,signs,_) in enumerate(attempt): # iterate over each action in an attempt
-                        if vid == 0: # if first action
-                            if 'Sink' in vname: # if sink, make dotted
-                                lines.append((0,'"'+translate(vname)+'" [style="dotted,filled", fillcolor= yellow]'))
+            for team_attacker, attempts in team_attack_attempts.items():  # For each attacker that obtains the objective
+                color = tcols[team_attacker.split('-')[0]]  # Team color
+                # _print_unique_attempts(team_attacker, attempts)
+
+                for attempt in attempts:
+                    # Record signatures
+                    for action in attempt:  # action = (vert_name, start_time, end_time, signs, timestamps)
+                        if action[0] not in node_signatures.keys():
+                            node_signatures[action[0]] = set()
+                        node_signatures[action[0]].update(action[3])
+
+                    # Create nodes for the AG
+                    for vid, action in enumerate(attempt):  # Iterate over each action in an attempt
+                        vname = action[0]
+                        if vid == 0:  # If first action
+                            if 'Sink' in vname:  # If sink, make dotted TODO: this check is always False
+                                lines.append('"' + _translate(vname) + '" [style="dotted,filled", fillcolor= yellow]')
```

Comment:
This is not executed (see comment above)

Reply:
How come?

Reply:
If vid is 0, that means it is the first action in a path. The code checks whether it is a sink or not. based on that, it sets the style of the vertex in the graph. Then comes the else condition (which is true for all the other actions in the path). Same process: checks if the action is in Sink, then set the style appropriately. Are you sure your indenting is correct?

Reply:
I only meant that `if 'Sink' in vname:` block is not executed, since as I have mentioned above, there are no states with 'Sink' in their name.